### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by jeetsdev**
## Added authentication

- Users can Login/Signup
- Cart and wishlist management using auth

[preview]()

### CODE REVIEW

**Review by shraddha15-creator — APPROVED** [review]
Liked that you used comments to understand the code part.
Easy to read and understand

File: src/pages/Auth/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    // password visibilty handler here 
+    const passVisibiltyHandler = () => {
+        return loginFormData.passType === "password" ? setloginFormData({ ...loginFormData, passType: "text" }) : setloginFormData({ ...loginFormData, passType: "password" });
+    }
+
+    // Checking if alredy login
+    useEffect(() => {
+        // 
+        if (authToken) { navigate(`${location?.state?.from?.pathname}`) }
```
Comment (Rohanmond) [inline src/pages/Auth/Login.jsx]: give threshold value like '/' , what if you location.state.from is undefined

File: src/pages/Auth/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        signUpFormHandler(signupFormData);
+    };
+
+    // password visibilty handler here 
+    const passVisibiltyHandler = () => {
+        return signupFormData.passType === "password" ? setsignupFormData({ ...signupFormData, passType: "text" }) : setsignupFormData({ ...signupFormData, passType: "password" });
+    }
+
+    useEffect(() => {
+        if (authToken) { navigate('/products') }
```
Comment (Rohanmond) [inline src/pages/Auth/SignUp.jsx]: here also you can provide dynamic page navigate

**Review by Rohanmond — APPROVED** [review]
you can go ahead

**Review by znjs — APPROVED** [review]

File: src/pages/Auth/Login.jsx (hunk 1 above)
Comment (jeetsdev) [reply src/pages/Auth/Login.jsx]: Missed that. Fixed now.

File: src/pages/Auth/SignUp.jsx (hunk 2 above)
Comment (jeetsdev) [reply src/pages/Auth/SignUp.jsx]: Yeah sure.

**Pull request merged**